pallets/pallet-xcm-benchmarks/src/fungible/benchmarking.rs

```diff
@@ -0,0 +1,242 @@
+				<
+					T::TransactAsset
+					as
+					Inspect<T::AccountId>
+				>::Balance::max_value() / 2u32.into(),
+			)?;
+		}
+
+		let assets: MultiAssets = vec![ teleportable_asset ].into();
+
+		let mut executor = new_executor::<T>(trusted_teleporter);
+		let instruction = Instruction::ReceiveTeleportedAsset(assets.clone());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm).map_err(|_| {
+			BenchmarkError::Override(
+				BenchmarkResult::from_weight(T::BlockWeights::get().max_block)
+			)
+		})?;
+	} verify {
+		assert!(executor.holding.ensure_contains(&assets).is_ok());
+	}
+
+	deposit_asset {
+		let asset = T::get_multi_asset();
+		let mut holding = T::worst_case_holding();
+
+		// Add our asset to the holding.
+		holding.push(asset.clone());
+
+		// our dest must have no balance initially.
+		let dest_location = T::valid_destination()?;
+		let dest_account = T::AccountIdConverter::convert(dest_location.clone()).unwrap();
+		assert!(T::TransactAsset::balance(&dest_account).is_zero());
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+		let instruction = Instruction::<XcmCallOf<T>>::DepositAsset {
+			assets: asset.into(),
+			max_assets: 1,
+			beneficiary: dest_location,
+		};
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// dest should have received some asset.
+		assert!(!T::TransactAsset::balance(&dest_account).is_zero())
+	}
+
+	deposit_reserve_asset {
+		let asset = T::get_multi_asset();
+		let mut holding = T::worst_case_holding();
+
+		// Add our asset to the holding.
+		holding.push(asset.clone());
+
+		// our dest must have no balance initially.
+		let dest_location = T::valid_destination()?;
+		let dest_account = T::AccountIdConverter::convert(dest_location.clone()).unwrap();
+		assert!(T::TransactAsset::balance(&dest_account).is_zero());
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+		let instruction = Instruction::<XcmCallOf<T>>::DepositReserveAsset {
+			assets: asset.into(),
+			max_assets: 1,
+			dest: dest_location,
+			xcm: Xcm::new(),
+		};
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// dest should have received some asset.
+		assert!(!T::TransactAsset::balance(&dest_account).is_zero())
+	}
+
+	initiate_teleport {
+		let asset = T::get_multi_asset();
+		let mut holding = T::worst_case_holding();
+
+		// Add our asset to the holding.
+		holding.push(asset.clone());
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+		let instruction = Instruction::<XcmCallOf<T>>::InitiateTeleport {
+			assets: asset.into(),
+			dest: T::valid_destination()?,
+			xcm: Xcm::new(),
+		};
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// we're good as long as there's no error, checking account is bypassed when operating on
+		// native assets aka Dot
```

Comment:
The absence of the check here is the only difference of `cumulus/pallet-xcm-benchmarks` against `polkadot/pallet-xcm-benchmarks`, right?

Is it necessary to have this pallet? Cant we reuse the one from polkadot?
It will be harder to maintain this. 
I think the general approach is to have the benchmarks next to the pallet itself (like unit tests), and whenever there is a change in the pallet itself, we need to make sure the tests are still valid and upgrade them as well.

Reply:
I have discussed this with @gilescope and we decided to port the pallet as the cumulus stuff and polkadot code are going to divulge further and maintaining those differences is more a pain than a win.

Reply:
Let's see how this goes and try to consolidate this back into the original repo if the need arises.